lighthouse-cli/test/smokehouse/test-definitions/forms/form-expectations.js

```diff
@@ -0,0 +1,291 @@
+ */
+'use strict';
+
+/**
+ * @type {Array<Smokehouse.ExpectedRunnerResult>}
+ * Expected Lighthouse artifacts from Form gatherer
+ */
+const expectations = [
+  {
+    artifacts: {
+      FormElements: [
+        {
+          attributes: {
+            id: 'checkout',
+            name: 'checkout',
+            autocomplete: 'on',
+            nodeLabel: 'Name on card: \nCredit card number: \nExpiry Date: \nMM\n01\n02\n03\n04\n05\n06\n07\n08\n09…',
+            snippet: '<form id="checkout" name="checkout" action="../done.html" method="post">',
+          },
+          inputs: [
+            {
+              id: 'name_cc1',
+              name: 'name_cc1',
+              placeholder: 'John Doe',
+              autocomplete: 'cc-name',
+              nodeLabel: 'textarea',
+              snippet: '<textarea type="text" id="name_cc1" name="name_cc1" autocomplete="cc-name" placeholder="John Doe">',
+            },
+            {
+              id: 'CCNo1',
+              name: 'CCNo1',
+              placeholder: '5555 5555 5555 5555',
+              autocomplete: 'cc-number',
+              nodeLabel: 'input',
+              snippet: '<input type="text" id="CCNo1" name="CCNo1" autocomplete="cc-number" placeholder="5555 5555 5555 5555">',
+            },
+            {
+              id: 'CCExpiresMonth1',
+              name: 'CCExpiresMonth1',
+              autocomplete: 'cc-exp-month',
+              nodeLabel: 'MM\n01\n02\n03\n04\n05\n06\n07\n08\n09\n10\n11\n12',
+              snippet: '<select id="CCExpiresMonth1" name="CCExpiresMonth1" autocomplete="cc-exp-month">',
+            },
+            {
+              id: 'CCExpiresYear1',
+              name: '',
+              autocomplete: 'cc-exp-year',
+              nodeLabel: 'YY\n2019\n2020\n2021\n2022\n2023\n2024\n2025\n2026\n2027\n2028\n2029',
+              snippet: '<select id="CCExpiresYear1" autocomplete="cc-exp-year">',
+            },
+            {
+              id: 'cvc1',
+              name: 'cvc1',
+              placeholder: '555',
+              autocomplete: 'cc-csc',
+              nodeLabel: 'input',
+              snippet: '<input id="cvc1" name="cvc1" autocomplete="cc-csc" placeholder="555">',
+            },
+          ],
+          labels: [
+            {
+              for: 'name_cc1',
+              nodeLabel: 'Name on card:',
+              snippet: '<label for="name_cc1">',
+            },
+            {
+              for: 'CCNo1',
+              nodeLabel: 'Credit card number:',
+              snippet: '<label for="CCNo1">',
+            },
+            {
+              for: 'CCExpiresMonth1',
+              nodeLabel: 'Expiry Date:',
+              snippet: '<label for="CCExpiresMonth1">',
+            },
+            {
+              for: 'cvc1',
+              nodeLabel: 'CVC:',
+              snippet: '<label for="cvc1">',
+            },
+          ],
+        },
+        {
```

Comment:
wasn't there an `attributes: undefined` here before? that plus a comment saying "this is the formless object" would be nice.